packages/cli/src/cmds/stats/stats.ts

```diff
@@ -0,0 +1,258 @@
+    }
+
+    async function showStats() {
+      UI.status = `Computing AppMap stats...`;
+      let appMapDir = directoryToUse;
+      let appMapSizes = await calculateAppMapSizes(appMapDir);
+      let sortedAppMapSizes = await sortAppMapSizes(appMapSizes);
+
+      UI.status = `Displaying biggest appmaps...\n`;
+      let biggestAppMapSizes: {size: number,
+                               name: string}[] = [];
+      let minAppMapSize = 1024*1024/2;
+      sortedAppMapSizes.filter((appmap) => appmap.size > minAppMapSize).slice(0, limitToUse).forEach((appmap) => {
+        biggestAppMapSizes.push({
+          size: appmap.size,
+          name: appmap.name,
+        });
+      });
+      if (json) {
+        console.log(JSON.stringify(biggestAppMapSizes));
+      } else {
+        biggestAppMapSizes.forEach((appmap) => {
+          console.log(sizeInMB(appmap.size) + 'MB ' + appmap.name)
+        });
+      }
+
+      UI.status = `Computing functions with highest AppMap overhead...\n`;
+      let executionTimes = await calculateExecutionTimes(appMapDir);
+      let sortedExecutionTimes = await sortExecutionTimes(executionTimes);
+      UI.status = `Displaying functions with highest AppMap overhead...\n`;
+
+      // column names in JSON files use snakecase
+      let slowestExecutionTimes: {elapsed_instrumentation_time_total: number,
+                                  num_calls: number,
+                                  name: string}[] = [];
+
+      // if there are no instrumentation data don't show this report
+      if (sortedExecutionTimes.length > 0 && sortedExecutionTimes[0].elapsedInstrumentationTimeTotal == 0) {
+        console.log('No "elapsed_instrumentation" data in the AppMaps.');
+      } else {
+        sortedExecutionTimes.slice(0, limitToUse).forEach((executionTime) => {
+          slowestExecutionTimes.push({
+            elapsed_instrumentation_time_total: Number(executionTime.elapsedInstrumentationTimeTotal.toFixed(6)),
+            num_calls: executionTime.numberOfCalls,
+            name: executionTime.name,
+          });
+        });
+        if (json) {
+          console.log(JSON.stringify(slowestExecutionTimes));
+        } else {
+          slowestExecutionTimes.forEach((executionTime) => {
+            console.log(String(executionTime.elapsed_instrumentation_time_total).padEnd(10, '0') + 's ' + executionTime.name + ' ' + executionTime.num_calls + ' calls');
+          });
+        }
+      }
+
+      let telemetryMetrics = {}
+      let telemetryMetricsCounter = 1;
+      biggestAppMapSizes.forEach((appmap) => {
+        telemetryMetrics[`biggestAppmaps_${telemetryMetricsCounter}`] = appmap.size
+        telemetryMetricsCounter += 1;
+      });
+      telemetryMetricsCounter = 1;
+      slowestExecutionTimes.forEach((executionTime) => {
+        telemetryMetrics[`slowestInstrumentationTimes_${telemetryMetricsCounter}`] = executionTime.elapsed_instrumentation_time_total;
```

Comment:
Would it be useful for us to see the actual function names? Then we could see if there are some common library functions that are always slow.

Reply:
Added.